Frontend/ui-customer/src/Components/Referral/View.purs

```diff
@@ -0,0 +1,189 @@
+
+  is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+  or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+  the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Components.Referral.View where
+
+import Components.Referral.Controller
+import Screens.Types (ReferralComponentState, ReferralStage(..))
+import Effect (Effect)
+import Prelude
+import Font.Size as FontSize
+import Font.Style as FontStyle
+import Styles.Colors as Color
+import PrestoDOM
+import PrestoDOM.Properties
+import Language.Strings (getString)
+import Language.Types (STR(..))
+import Prelude ((<>))
+import Engineering.Helpers.Commons as EHC
+import PrestoDOM.Types.DomAttributes
+import Common.Types.App (LazyCheck(..))
+import JBridge as JB
+import PrestoDOM.Animation as PrestoAnim
+import Animation (translateYAnim,translateYAnimFromTop, fadeIn)
+import Animation.Config (translateYAnimConfig)
+import Helpers.Utils (FetchImageFrom(..), fetchImage)
+import Components.PopUpModal as PopUpModal
+import Data.String (length)
+import Components.PrimaryButton as PrimaryButton
+import Mobility.Prelude
+import Data.Array (any)
+import ConfigProvider
+
+view :: forall w. (Action -> Effect Unit) -> ReferralComponentState -> PrestoDOM (Effect Unit) w
+view push state =
+  relativeLayout
+  [ width MATCH_PARENT
+  , height MATCH_PARENT
+  , adjustViewWithKeyboard "true"
+  , clickable true
+  ][ enterReferralCodeView push state
+   , referralPopUp push state
+   , referralInfoPopup push state
+   ]
+
+
+enterReferralCodeView :: forall w. (Action -> Effect Unit) -> ReferralComponentState -> PrestoDOM (Effect Unit) w
+enterReferralCodeView push state =
+  PrestoAnim.animationSet [ translateYAnim translateYAnimConfig ] $
+  linearLayout
+  [ height MATCH_PARENT
+  , width MATCH_PARENT
+  , gravity CENTER
+  , orientation VERTICAL
+  , clickable true
+  , background Color.blackLessTrans
+  , visibility $ boolToVisibility $ state.stage == ENTER_REFERRAL_CODE && state.stage /= NO_REFERRAL_STAGE
+  ][  linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , cornerRadius 16.0
+      , background Color.white900
+      , orientation VERTICAL
+      , gravity CENTER_HORIZONTAL
+      , margin $ Margin 24 24 24 15
+      , padding $ Padding 16 16 16 8
+      , alignParentBottom "true,-1"
+      , clickable true
+      ][  imageView
+          [ height $ V 160 
+          , width $ V $ (EHC.screenWidth unit) - 20
+          , imageWithFallback $ fetchImage FF_COMMON_ASSET "ny_ic_refer_friend"
+          , cornerRadius 10.0
+          ]
+        , linearLayout
+          [ height WRAP_CONTENT
+          , width WRAP_CONTENT
+          , margin $ MarginTop 10
```

Comment:
instead of marginTop, can we have paddingTop? So that click area will be more otherwise as I can see it's just text view currently

Reply:
also, can ripple be added here as well?